src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -195,5 +196,33 @@ public override async Task<VSCodeAction[]> ProvideCodeActionsAsync(CodeActionPar
 
             return results;
         }
+
+        public override async Task<SemanticTokens> ProvideSemanticTokensAsync(SemanticTokensParams semanticTokensParams, CancellationToken cancellationToken)
```

Comment:
Needs testing, should just be able to copy over the test cases from `ProvideCodeActionsAsync` 😄